### PR TITLE
feat: refresh once on navigation & es build

### PR DESCRIPTION
**Pull request by pimlie**
Also some more optimisation and refactoring.

The `refreshOnceOnNavigation` option is a workaround for #320, but if you have a lot of components that set metaInfo it could also be a performance improvement. Maybe we should just make it default `true`?

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Cleanup
+rm -rf lib es
+
+echo 'Compile JS...'
+rollup -c scripts/rollup.config.js
+echo 'Done.'
+echo ''
+
+echo 'Build ES modules...'
+NODE_ENV=es babel src --out-dir es --ignore 'src/browser.js'
```
Comment (manniL) [inline scripts/build.sh]: I'd rather move that back to node scripts. We can't assume/test that'll work on all OS, especially windows.

Comment (pimlie) [reply scripts/build.sh]: Sure, np. I shamelessly copied this from bootstrap-vue as that also has a build script by @pi0 :tm:

Comment (pimlie) [reply scripts/build.sh]: Done, albeit under protest because bash is the best!

File: src/shared/ensure.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import isArray from './isArray'
+import { isObject } from './typeof'
+
+export function ensureIsArray(arg, key) {
+  if (key && isObject(arg)) {
+    if (!isArray(arg[key])) {
+      arg[key] = []
+    }
+    return arg
+  } else {
```
Comment (manniL) [inline src/shared/ensure.js]: Invert if/else and using early return might improve code clarity here ☺️

Comment (pimlie) [reply src/shared/ensure.js]: Well, its actually a bit of a strange function. I added it to ensure that the property of an object is an array as thats what I needed for allthose lifecycle hooks. Then I thought lets make it a bit more generic, even tough I dont need it.

This is the best I could make of it. We could change the `if(!isArray)` statement recursive (`arg[key] = ensureIsArray(arg[key])`) which might makes it more readable but I dont like that its costs an extra function call and/or that we always set the value for arg[key]

File: src/client/triggerUpdate.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import batchUpdate from './batchUpdate'
+
+// store an id to keep track of DOM updates
+let batchId = null
+
+export default function triggerUpdate(vm, hookName) {
+  if (vm.$root._vueMetaInitialized && !vm.$root._vueMetaPaused) {
```
Comment (pimlie) [inline src/client/triggerUpdate.js]: What do you think about grouping these properties in a single `_vuemeta` object? Then the `_vuemeta` property will be an object on `$root` and a boolean on all childNodes of $root (so it also replaced `_hasMetaInfo`, but that could be a breaking change althought its only  meant to be used internally)

Comment (manniL) [reply src/client/triggerUpdate.js]: Like the idea, yup!

Comment (manniL) [reply src/client/triggerUpdate.js]: About the breaking change, that's true. And might influence things like `nuxt-i18n`. We should def. point that out

File: src/shared/ensure.js (hunk 2 above)
Comment (manniL) [reply src/shared/ensure.js]: From
```js
export function ensureIsArray(arg, key) {
  if (key && isObject(arg)) {
    if (!isArray(arg[key])) {
      arg[key] = []
    }
    return arg
  } else {
    return isArray(arg) ? arg : []
  }
}
```

to

```js
export function ensureIsArray(arg, key) {
  if (!key || !isObject(arg)) {
    return isArray(arg) ? arg : []
  }

  if (!isArray(arg[key])) {
    arg[key] = []
  }
  return arg
}
```

Comment (pimlie) [reply src/shared/ensure.js]: Do you really think thats better readable then 4968ee10b92864b0dd8e4eb7a3d52612553d767a? In my mind the `||` with the inline if makes me have to read that part twice to understand whats happening, while 4968ee10b92864b0dd8e4eb7a3d52612553d767a reads much more fluently.

Comment (manniL) [reply src/shared/ensure.js]: @pimlie `!(key && isObject(arg))` would be fine for me as well. :relaxed:

Comment (pimlie) [reply src/shared/ensure.js]: Yeah but thats goes against the principle that `if(true)` provides better readability then `if(!false)` ;)

I have changed it to your proposal, even though I dont think its more readable I also dont care enough to have lengthy discussions about it ;)

File: src/client/triggerUpdate.js (hunk 3 above)
Comment (pimlie) [reply src/client/triggerUpdate.js]: Wdyt about this solution, now I provide a `hasMetaInfo()` export which external libraries could and should use to check if meta info has been defined. That way we can keep the exact implementation for that internally (like _hasMetaInfo was meant to be)

Comment (pimlie) [reply src/client/triggerUpdate.js]: Hmm, this works for the es module but not for cjs or umd due to combining default and named exports ofc

Comment (pimlie) [reply src/client/triggerUpdate.js]: Would you be ok with removing the single / default export and only have named exports? Its a good reason to release 2.0 ;)

Comment (manniL) [reply src/client/triggerUpdate.js]: @pimlie I'd be fine with that!

Comment (pimlie) [reply src/client/triggerUpdate.js]: I changed it to that but didnt like it. So now I just added a default object export. Looked a bit around and this seems to be the more used solution by others vue plugins as well (they either do this or export a class with a static install method, the latter we might use for 3.0)

**Comment by pimlie** [comment]
This is ready for review again. Please merge when ok so I can start working on optimizing getMetaInfo and e2e tests

**Pull request merged**